pages/admin/rollup.js

```diff
@@ -0,0 +1,174 @@
+  const buttonRef = useRef()
+
+  const handleOpenDialog = event => {
+    if (buttonRef.current) {
+      buttonRef.current.open(event)
+    }
+  }
+
+  const handleOnFileLoad = data => setCsvData(mapCoordinapeData(data))
+
+  const handleRemoveFile = event => {
+    if (buttonRef.current && csvData) {
+      buttonRef.current.removeFile(event)
+      setCsvData(null)
+    }
+  }
+
+  const handleSubmit = async () => {
+    try {
+      setLoading(true)
+      setMembersData(null)
+      const res = await fetch('/api/harness', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(csvData),
+      })
+
+      const { updatedData } = await res.json()
+      setMembersData(updatedData)
+      setLoading(false)
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  const handleRollup = () => {
+    console.log('rollup!')
+  }
+
+  const ipfsUrl =
+    membersData && membersData.cid
+      ? `https://ipfs.io/ipfs/${membersData.cid}/level.json`
+      : null
+
+  return (
+    <div>
+      <Head>
+        <title>Level Protocol - Rollup</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <main>
+        <Public>
+          <MenuBar>
+            <Link href="/about">About</Link>
+            <Link href="/join">Join</Link>
+          </MenuBar>
+          <PageContent>
+            <LevelWindow
+              backgroundColor="vibrantBlack"
+              enableActions={false}
+              maxHeight="75vh"
+              title="Rollup"
+            >
+              <Article>
+                <DashboardWrapper>
+                  <H2>Submit Members Data</H2>
+                  {ipfsUrl ? (
+                    <>
+                      <IpfsLink>
+                        <a href={ipfsUrl} target="_blank" rel="noreferrer">
+                          See it on IPFS!
+                        </a>
+                      </IpfsLink>
+
+                      <H4 color="white">Submit to LVL</H4>
+                      <Button onClick={handleRollup}>Rollup</Button>
+                    </>
+                  ) : (
+                    <>
+                      <FileUploader
+                        title="Upload your Coordinape Data File"
+                        buttonRef={buttonRef}
+                        handleOnFileLoad={handleOnFileLoad}
+                        handleRemoveFile={handleRemoveFile}
+                        handleOpenDialog={handleOpenDialog}
+                      />
+                      <H4 color="white">Update members data</H4>
+                      <Button onClick={handleSubmit}>Submit</Button>
+                      {loading && !ipfsUrl && <Spinner />}
+                    </>
+                  )}
+                </DashboardWrapper>
+              </Article>
+            </LevelWindow>
+          </PageContent>
+        </Public>
+      </main>
+    </div>
+  )
+}
+
+export default Converter
```

Comment:
I'm realizing we are using the term "rollup" for 2 different things. We might want to call this "integration" so we can reserve the concept of rollups for skill rollup functions, do you agree?

We could potentially structure it like:
`/dao` - dashboard showing their dao information, name, address, list of registered skills, etc.
`/dao/settings` - config view that lets them set their dao information, name, address, etc.
`/dao/register-skill-set` - form for registering a new skill/set
`/dao/configure-skill-set` - form for configuring a skill set, ie. naming the 32 skills within the skillset
`/dao/integrations` - list of potential integrations like coordinape, sourcecred, etc. that they can configure each one and turn them on or off (they probably start off by default)
`/dao/integrations/[id]` - detail/config view for a particular integration, ie. `/dao/integrations/coordinape` lets them configure their coordinape instance path, any keys, etc.
`/dao/skills` - list of their registered skills
`/dao/skills/[id]` - detail/config view for a particular skill allowing them to configure which rollup functions to use for that skill